### ISSUE TITLE
Feedback from `BadPrecondition` is hard to understand

### ISSUE THREAD

**Issue by abailly-iohk** [input-output-hk/quickcheck-dynamic#19]
I finally got back to working on using DL formulas to test Hydra, and wrote this small property:
```
conflictFreeLiveness :: DL WorldState ()
conflictFreeLiveness = do
  anyActions_
  getModelStateDL >>= \case
    st@WorldState{hydraState = Open{}} -> trace ("generating payment in state " <> show st) $ do
      (party, payment) <- forAllQ (nonConflictingTx st)
      action $ Model.NewTx party payment
      eventually (ObserveConfirmedTx payment)
    _ -> pass
 where
  nonConflictingTx st = withGenQ (genPayment st) (const [])
  eventually a = anyActions 10 >> action (Wait 10) >> action a
```
(See [here](https://github.com/input-output-hk/hydra-poc/blob/ensemble/implement-liveness-in-DL/hydra-node/test/Hydra/ModelSpec.hs#L118) for more context)

The test fails with a `BadPrecondition` error which I fail to understand: The way `genPayment` is defined, it indirectly checks the `precondition` holds by generating only a transaction from the confirmed UTxO set. 
There's probably something I am failing to understand in how the DL is run.

**Comment by abailly-iohk** [comment]
An alternative "error" I am "seeing" is that the test hangs and it's hard to know what it is doing.

**Comment by MaximilianAlgehed** [comment]
Yes, I think its worth doing something about that. Can make an issue saying
we need better errors?

**Comment by abailly-iohk** [comment]
That's the intention of this issue

**Comment by MaximilianAlgehed** [comment]
This is fixed by #28.
